modules/deployment-model-impl/src/test/java/org/mule/runtime/module/deployment/impl/internal/builder/ArtifactPluginFileBuilder.java

```diff
@@ -121,6 +143,19 @@ public ArtifactPluginFileBuilder dependingOn(String pluginName) {
       customResources.add(new ZipResource(applicationPropertiesFile.getAbsolutePath(), PLUGIN_PROPERTIES));
     }
 
+    if (mulePluginModel != null) {
+      final File jsonDescriptorFile = new File(getTempFolder(), META_INF + separator + MULE_PLUGIN_JSON);
+      jsonDescriptorFile.deleteOnExit();
+
+      String jsonDescriber = new MulePluginModelJsonSerializer().serialize(mulePluginModel);
+      try {
+        FileUtils.writeStringToFile(jsonDescriptorFile, jsonDescriber);
+      } catch (IOException e) {
+        throw new IllegalStateException("There was an issue generating the JSON file for " + this.getId(), e);
+      }
+      customResources.add(new ZipResource(jsonDescriptorFile.getAbsolutePath(), META_INF + "/" + MULE_PLUGIN_JSON));
```

Comment:
`File.separator` won't work here unless I'm mistaken.
The value in a unix system will be "/", but in Windows' based will be "\", and as far as I understand the `ZipResource#alias` is used to generate a `ZipEntry`, which relies only in "/".

Is the above correct? if so, I'll leave it as it is